src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -1614,6 +1614,59 @@ class spell_gen_elune_candle : public SpellScript
     }
 };
 
+// 50051 - Ethereal Pet Aura
+enum EtherealPet
+{
+    SPELL_STEAL_ESSENCE_VISUAL          = 50101,
+    SPELL_CREATE_TOKEN                  = 50063
+};
+
+class spell_ethereal_pet_aura : public AuraScript
+{
+    PrepareAuraScript(spell_ethereal_pet_aura);
+
+    void OnProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        PreventDefaultAction();
+
+        Player* owner = GetTarget()->ToPlayer();
+        Unit* target = eventInfo.GetActionTarget();
+        if (!owner || !target)
+            return;
+
+        Creature* pet = owner->GetMap()->GetCreature(owner->m_SummonSlot[SUMMON_TYPE_MINIPET]);
```

Comment:
i dont like it.
You can iterate player pets instead

Reply:
Why? Isn't it better to access directly?

Reply:
> iterate player pets instead
that

Reply:
How should I determine if pet is actually summoned minipet in such case?

Reply:
you should use SUMMON_SLOT_MINIPET instead of SUMMON_TYPE_MINIPET.. and the other thing.. the only thing you could try is, accessing TempSummons where you have access on m_properties, but honestly? i think accessing it like you did, is good enough. Can't see a reason why you shouldn't do that.

Reply:
but what you should do is, check if you even got a pet in this slot.. you don't check if its null or not

Reply:
Added check here.

Reply:
```cpp
void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
{
    PreventDefaultAction();
    for (Unit* soulTrader : GetUnitOwner()->m_Controlled)
        if (soulTrader->GetEntry() == NPC_ETHEREAL_SOUL_TRADER)
        {
            soulTrader->CastSpell(eventInfo.GetProcTarget(), SPELL_ETHEREAL_STEAL_ESSENCE);
            return;
        }
}
```